playback/tape_cassettes/s3/s3_tape_cassette.py

```diff
@@ -242,18 +243,30 @@ def content_filter_func(recording_str):
                 return True
             content_filter = content_filter_func
 
-        for key in self._s3_facade.iter_keys(
-                prefix=self.METADATA_KEY.format(key_prefix=self.key_prefix, id='{}/'.format(category)),
-                start_date=start_date,
-                end_date=end_date,
-                content_filter=content_filter,
-                limit=limit):
-            result = self._metadata_key_parser.parse(key)
-            if result is None:
-                continue
-            recording_id = result.named['id']
-            _logger.info(u'Found filtered recording id {}'.format(recording_id))
-            yield recording_id
+        # to improve performance when looking for recordings in s3, the date is added to the folder
+        # and when a start date is given we can look for specific folders until today (or end_time)
+        if start_date:
+            end_date = end_date or datetime.utcnow()
+            days = [(start_date + timedelta(days=i)) for i in range((end_date - start_date).days + 1)]
+            id_prefixes = ['{}/{}/'.format(category, day.strftime('%Y%m%d')) for day in days]
+            start_date = None
```

Comment:
Why reset it instead of not passing it to the facade at all?
also what if end_date is not None and start_date is? you will still pass end_date

Reply:
if you only have an end date, then I search the entire bucket and don't go to specific dates, so I need the end date in for the last_modified predicate.
regarding setting it to None, you can see that this is way less code, and does exactly that. otherwise I'd need to ask again if start_date is set and in that case not send start_date and end_date which is a bit convoluted.